private/aws-restjson-server/src/protocols/Aws_restJson1.ts

```diff
@@ -2053,7 +2053,7 @@ export const deserializeMalformedContentTypeWithoutBodyEmptyInputRequest = async
   );
   if (contentTypeHeaderKey != null) {
     const contentType = output.headers[contentTypeHeaderKey];
-    if (contentType !== undefined && contentType !== "application/json") {
```

Comment:
It's probably here: https://github.com/smithy-lang/smithy-typescript/blob/main/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java#L1871